src/Tokenizer/TokensAnalyzer.php

```diff
@@ -499,81 +490,72 @@ public function isUnarySuccessorOperator(int $index): bool
 
         $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
 
-        return $prevToken->equalsAny($allowedPrevToken);
+        return $prevToken->equalsAny([
+            ']',
+            [\T_STRING],
+            [\T_VARIABLE],
+            [CT::T_ARRAY_INDEX_CURLY_BRACE_CLOSE],
+            [CT::T_DYNAMIC_PROP_BRACE_CLOSE],
+            [CT::T_DYNAMIC_VAR_BRACE_CLOSE],
+        ]);
     }
 
     /**
      * Checks if there is a unary predecessor operator under given index.
      */
     public function isUnaryPredecessorOperator(int $index): bool
     {
-        static $potentialSuccessorOperator = [\T_INC, \T_DEC];
-
-        static $potentialBinaryOperator = ['+', '-', '&', [CT::T_RETURN_REF]];
-
-        static $otherOperators;
-
-        if (null === $otherOperators) {
-            $otherOperators = ['!', '~', '@', [\T_ELLIPSIS]];
-        }
-
-        static $disallowedPrevTokens;
-
-        if (null === $disallowedPrevTokens) {
-            $disallowedPrevTokens = [
-                ']',
-                '}',
-                ')',
-                '"',
-                '`',
-                [CT::T_ARRAY_SQUARE_BRACE_CLOSE],
-                [CT::T_ARRAY_INDEX_CURLY_BRACE_CLOSE],
-                [CT::T_DYNAMIC_PROP_BRACE_CLOSE],
-                [CT::T_DYNAMIC_VAR_BRACE_CLOSE],
-                [\T_CLASS_C],
-                [\T_CONSTANT_ENCAPSED_STRING],
-                [\T_DEC],
-                [\T_DIR],
-                [\T_DNUMBER],
-                [\T_FILE],
-                [\T_FUNC_C],
-                [\T_INC],
-                [\T_LINE],
-                [\T_LNUMBER],
-                [\T_METHOD_C],
-                [\T_NS_C],
-                [\T_STRING],
-                [\T_TRAIT_C],
-                [\T_VARIABLE],
-            ];
-        }
-
         $tokens = $this->tokens;
         $token = $tokens[$index];
 
-        if ($token->isGivenKind($potentialSuccessorOperator)) {
+        if ($token->isGivenKind([\T_INC, \T_DEC])) {
```

Comment:
the var names we checked here were descriptive and meaningful.

now, logic is the same, but those descriptions are gone.
maybe move the to comment ?